packages/lit-html/src/lit-html.ts

```diff
@@ -653,6 +653,22 @@ const getTemplateHtml = (
   const htmlResult: string | TrustedHTML =
     html + (strings[l] || '<?>') + (type === SVG_RESULT ? '</svg>' : '');
 
+  // A security check to prevent spoofing of Lit template results.
+  // In the future, we may be able to replace this with Array.isTemplateObject,
+  // though we might need to make that check inside of the html and svg
+  // functions, because precompiled templates don't come in as
+  // TemplateStringArray objects.
+  if (!Array.isArray(strings) || !strings.hasOwnProperty('raw')) {
```

Comment:
There are two other potentially reasonable places to do this:
  1. In the Template constructor, which will be called the same number of times as getTemplateHtml
  2. In `ChildPart._$setValue`, which is called more often but where our actual runtime duck-typing lives and might indicate other type-checks that need to be tightened. The nice this about this location is that we wouldn't need to throw, we could just fall through to generic object handling.

Either of those would leave getTemplateHtml open for use with arrays of strings for other utility purposes. I think it's worth ruling them out at least before committing.

Reply:
> Either of those would leave getTemplateHtml open for use with arrays of strings for other utility purposes

getTemplateHtml is a core part of the security boundary, and we should think very carefully before calling it with something other than a TemplateStringsArray. 

For `ChildPart._$setValue`, I think it's fine to continue to disallow directly rendering objects with '_$lit...$' keys.

The thing I like about doing it here is that, by default it's good to co-locate security checks with blessing the value as secure with policy.createHtml

If we change our minds later, I don't think it'd be a disruptive change to move this check elsewhere.

Reply:
Seem reasonable. I do wonder if the policy usage wouldn't also be good to put in the Template constructor, but it's effectively the same except for outside callers of `getTemplateHtml`.